src/partition/repart.c

```diff
@@ -3916,6 +3924,153 @@ static int context_mangle_partitions(Context *context) {
         return 0;
 }
 
+static int split_name_printf(Partition *p) {
+        assert(p);
+
+        const Specifier table[] = {
+                { 't', specifier_id128, &p->type_uuid },
+                { 'u', specifier_id128, &p->new_uuid  },
+
+                COMMON_SYSTEM_SPECIFIERS,
+                {}
+        };
+
+        return specifier_printf(p->split_name_format, NAME_MAX, table, arg_root, p, &p->split_name_resolved);
+}
+
+static int split_name_resolve(Context *context) {
+        int r;
+
+        LIST_FOREACH(partitions, p, context->partitions) {
+                if (p->dropped)
+                        continue;
+
+                if (!p->split_name_format)
+                        continue;
```

Comment:
I think it would make sense to default to the partition type for this if not specified explicitly. i.e. no reason to insist this is configured, if the partition type likely works for most people. I mean, by specifiying `--split` users made very clear they want this split, and I think there's value in just making that work with no conf if we can

Reply:
But what type do we use though? `root` and `root-x86_64` will evaluate to the same guid on x86-64 systems. We could either convert the final partition type back to its full name including arch, or we store the string type given by the user and reuse that here.

Reply:
hmm, not a fan of reusing the non-normalized string the user typed in. we should always normalize it.

But I wonder, is `root-x86-64` really that bad? Sounds pretty descriptive to me...

Right now when translating a uuid to a friendly name we use the arch specific one preferably i.e. the most precise. That sounds like would be ok here, too.

Reply:
What if users only want to split some partitions but not all? I figured by defaulting this to empty, only the partitions that SplitName= is explicitly set for are split out, and the others aren't. By doing this, we're defaulting to splitting every partition

Reply:
hmm, good point. But it I'd always assume that people would rather want an opt-out than an opt-in, after all they already asked explicit for `--split` on the command line...

I think we should just say that if people specify `SplitName=-` it's an opt-out. I mean, this would take the ability away for people to name their files `-`, but I don't see that as a big loss...

Reply:
Should we default to something that includes the partition number so that we don't fail when there are multiple partitions of the same type?